addons/web_editor/tools.py

```diff
@@ -97,6 +101,13 @@ def get_video_url_data(video_url, autoplay=False, loop=False, hide_controls=Fals
             params['controls'] = 0
         if loop:
             params['loop'] = 1
+        groups = platform_match.groupdict()
+        if 'hash' in groups:
+            params['h'] = groups['hash']
+        elif 'params' in groups:
+            url_params = parse_qs(groups['params'])
+            if 'h' in url_params:
+                params['h'] = url_params['h'][0]
```

Comment:
I understand you need to grab the hash for the unlisted video, but why do we need all the params?

Reply:
Here, I used `parse_qs` to: (1) **reliably** extract the parameter I want and (2) to keep the regular expression as simple as possible. To use the `parse_qs` function, I need to pass all the URL params.

The advantage of doing that is that, in the future, we can extract other URL parameters without needing to update the regular expression and make it more complex.

You would prefer creating a new group within the regular expression to extract the `h` parameter in order to avoid the override security?

Reply:
No I have enough context now, sounds good to me, thanks. You need to ask security to review it anyway though, I can't override it myself.